lib/ansible/modules/cloud/vmware/vmware_host_powerstate.py

```diff
@@ -0,0 +1,222 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2018, Abhijeet Kasurde <akasurde@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = r'''
+---
+module: vmware_host_powerstate
+short_description: Manages power states of host systems in vCenter
+description:
+- This module can be used to manage power states of host systems in given vCenter infrastructure.
+- User can set power state to 'power-down-to-standby', 'power-up-from-standby', 'shutdown-host' and 'reboot-host'.
+- State 'reboot-host', 'shutdown-host' and 'power-down-to-standby' are not supported by all the host systems.
+version_added: 2.6
+author:
+- Abhijeet Kasurde (@akasurde) <akasurde@redhat.com>
+requirements:
+- python >= 2.6
+- PyVmomi
+options:
+  state:
+    description:
+    - Set the state of the host system.
+    choices: [ 'power-down-to-standby', 'power-up-from-standby', 'shutdown-host', 'reboot-host' ]
+    default: 'shutdown-host'
+  esxi_hostname:
+    description:
+    - Name of the host system to work with.
+    - This is required parameter if C(cluster_name) is not specified.
```

Comment:
Not added in module definition, (required_one_of)

Reply:
Done.